src/compiler/scala/tools/nsc/classpath/ZipAndJarFileLookupFactory.scala

```diff
@@ -215,3 +242,26 @@ final class FileBasedCache[T] {
     cache.clear()
   }
 }
+
+object FileBasedCache {
+  private val deferCloseMs = Integer.getInteger("scalac.filebasedcache.defer.close.ms", 1000)
```

Comment:
The tension here is that too long a delay could lead to an error (on Windows) with an inabilty to overwrite the JAR. To short a delay and the entry could be evicted before a subsequent sub-project compilation is able to get a cache hit. A more comprehensive solution would be to involve build tools in the policy: they could close entries with refcount of zero when that entry's JAR is about to be overwritten.